### PR TITLE
fix: synchronize active tab state with URL to prevent mismatch after …

### PR DESCRIPTION
**Pull request by youngheart12**
…navigation

### What does this implement/fix? Explain your changes.
This fix resolves an issue where, upon navigating to a new page, the selected tab remained unchanged while the content updated correctly. The fix ensures that the active tab is synchronized with the URL, so the correct tab is always highlighted after page navigation. The state is now derived directly from the URL to prevent any mismatch between the selected tab and the displayed content.

...

### Does this close any currently open issues?
Yes this closes issue #2366 
...

### Any other comments?
None.
...

### Dependent PRs/Commits

...

### Describe breaking changes, if any.
No breaking changes introduced with this fix.
...

### Checklist

Check all those that are applicable and complete.

- [ ] Merged with latest `master` branch

### CODE REVIEW

File: docs/src/components/Container/ComponentsContainer.js [hunk 1]
```diff
@@ -40,9 +32,16 @@ const ComponentsContainer = ({ children, pageTitle, relativePagePath, tabs, page
     const pages = pagePath.slice(0, pagePath.length - 1);
     const path = `${pages.join('/')}/${nextTabSlug}/`;
     navigate(path, { state: { animation: false } });
-    setActiveIndex(tabIndex);
   };
 
+  const activeTabIndex = React.useMemo(() => {
+    const foundIndex = tabsList?.length
```
Comment (anuradha9712) [inline docs/src/components/Container/ComponentsContainer.js]: Please update this variable name `foundIndex` to be more appropriate

Comment (youngheart12) [reply docs/src/components/Container/ComponentsContainer.js]: ok

**Review by anuradha9712 — CHANGES_REQUESTED** [review]
Please submit a pull request with `develop` as the target branch.

**Comment by youngheart12** [comment]
@anuradha9712 this is updated,Ready for review

**Review by anuradha9712 — APPROVED** [review]

**Pull request merged**